identity/src/commonMain/kotlin/com/android/identity/nfc/NdefRecord.kt

```diff
@@ -87,19 +89,19 @@ data class NdefRecord(
         if (isShortRecord) flags = flags or FLAG_SR
         if (idLengthPresent) flags = flags or FLAG_IL
 
-        buf.writeByte(flags.toByte())
-        buf.writeByte(type.size.toByte())
+        bsb.appendUInt8(flags)
+        bsb.appendUInt8(type.size.toByte().toInt()) //TODO: b/393388370 - fails in tests without .toByte() which ignores higher byte of Int.
```

Comment:
What happens if you change this to just `bsb.appendUInt(type.size)`? That's the way it _should work_, I mean, we want application-level code using this to not have to use any of of these `toXxx()` functions.